implants/imix/imix_config.yaml.example

```diff
@@ -0,0 +1,182 @@
+# Imix Build Configuration Example
+#
+# All fields are optional. If not specified, the build process will use:
+# 1. Individual environment variables (e.g., IMIX_CALLBACK_URI)
+# 2. Default values
+#
+# Usage (container-friendly pattern):
+#   Set IMIX_BUILD_CONFIG environment variable with the YAML content:
+#
+#   export IMIX_BUILD_CONFIG="$(cat imix_config.yaml.example)"
+#   cargo build --release
+#
+# Or inline:
+#   export IMIX_BUILD_CONFIG='
+#   callback_uri: "http://tavern.example.com:8000"
+#   callback_interval: 10
+#   '
+#   cargo build --release
+#
+# In Dockerfile:
+#   ENV IMIX_BUILD_CONFIG="callback_uri: http://tavern.example.com:8000\ncallback_interval: 10"
+#
+# In docker-compose.yml:
+#   environment:
+#     IMIX_BUILD_CONFIG: |
+#       callback_uri: "http://tavern.example.com:8000"
+#       callback_interval: 10
+
+# ============================================================================
+# CALLBACKS CONFIGURATION
+# ============================================================================
+# You can specify callbacks in two formats:
+#
+# 1. NEW FORMAT (recommended): List of callbacks with individual settings
+# 2. LEGACY FORMAT: Single callback with global interval/retry settings
+#
+# The new format takes precedence if both are specified.
+
+# --- NEW FORMAT: Multiple Callbacks ---
+# Each callback can have its own URI, interval, and retry_interval
+callbacks:
+  # Primary callback - DNS transport with custom intervals
+  - uri: "dns://8.8.8.8:53?domain=c2.example.com&type=txt"
+    interval: 10          # Check every 10 seconds
+    retry_interval: 5     # Retry after 5 seconds on error
+
+  # Secondary callback - HTTP fallback with different timing
+  - uri: "http://tavern.example.com:8000"
+    interval: 30          # Check every 30 seconds
+    retry_interval: 15    # Retry after 15 seconds on error
+
+  # Tertiary callback - HTTPS with longer intervals
+  - uri: "https://backup.example.com:443"
+    interval: 60          # Check every 60 seconds
+    retry_interval: 30    # Retry after 30 seconds on error
+
+# --- LEGACY FORMAT: Single Callback (for backward compatibility) ---
```

Comment:
Remove legacy format we don't need this. Users can still set the IMIX_CALLBACK_URI env variable manually.

If both legacy vars and the config file var are set build.rs should error that you can't use both at the same time.